mllib/src/main/scala/org/apache/spark/mllib/nlp/CRF.scala

```diff
@@ -0,0 +1,265 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.nlp
+
+import org.apache.spark.annotation.DeveloperApi
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkContext
+import scala.collection.mutable.ArrayBuffer
+
+private[mllib] class CRF extends Serializable{
+  private val freq: Int = 1
+  private val maxiter: Int = 100000
+  private val cost: Double = 1.0
+  private val eta: Double = 0.0001
+  private val C: Float = 1
+  private val threadNum: Int = Runtime.getRuntime.availableProcessors()
+  private val threadPool: Array[CRFThread] = new Array[CRFThread](threadNum)
```

Comment:
Instead of using the thread pool, we should use the RDD right? so it's naturally in a multiple thread/processes model.